### PR TITLE
Add authentication to updater

### PR DESCRIPTION
**Pull request by LukasReschke**
This adds some authentication to the updater using a shared secret in `config.php`. The whole process is cookieless and thus if somebody closes the tab all authentication is gone (which is desired from a security PoV)

To test this define an `updater.secret` in `config.php` and try to access it with a valid or invalid token.

The next step is to add the automagic login from core once this is in. We can do that using the new updatenotification app at https://github.com/owncloud/core/pull/22238.

@VicDeo Please review.

### CODE REVIEW

File: app/bootstrap.php [hunk 1]
```diff
@@ -38,7 +38,5 @@
 	define('CURRENT_DIR', getcwd());
 }
 
-session_start();
```
Comment (VicDeo) [inline app/bootstrap.php]: @LukasReschke please leave this in place. I use session to store info between ajax requests. :)

**Comment by LukasReschke** [comment]
PS: The CSS sucks. But this is something that we or @jancborchardt can fix later :see_no_evil:

**Comment by VicDeo** [comment]
@LukasReschke as or me I just steal markup and assets from core :black_joker:

File: app/bootstrap.php (hunk 1 above)
Comment (LukasReschke) [inline app/bootstrap.php]: Back it is :smile:

**Comment by VicDeo** [comment]
@LukasReschke You're magician!
:+1:

**Pull request merged**